README.md
cm-web
===

The Awesome Web of Code Ministers


changelog.txt
== Changelog ==

= 0.4 =
* Added changelog.txt file.
* Updated Hybrid Core to 1.4.3.
* Tuned the Theme Settings page to adhere to the core WordPress admin styles.
* Fixed a jQuery bug preventing the Media Tools button from functioning in the Theme Settings.
* Improved Theme Settings - escaped option values and used selected() for select lists.
* Removed origin_entry_title_shortcode() from functions.php (not needed anymore).
* Added German translation.

footer.php
<?php
/**
 * Footer Template
 *
 * The footer template is generally used on every page of your site. Nearly all other
 * templates call it somewhere near the bottom of the file. It is used mostly as a closing
 * wrapper, which is opened with the header.php file. It also executes key functions needed
 * by the theme, child themes, and plugins. 
 *
 * @package Origin
 * @subpackage Template
 */
?>
				<?php get_sidebar( 'primary' ); // Loads the sidebar-primary.php template. ?>

				<?php do_atomic( 'close_main' ); // origin_close_main ?>

		</div><!-- #main -->

		<?php do_atomic( 'after_main' ); // origin_after_main ?>

		<?php get_sidebar( 'subsidiary' ); // Loads the sidebar-subsidiary.php template. ?>		

		<?php do_atomic( 'before_footer' ); // origin_before_footer ?>

		<div id="footer">

			<?php do_atomic( 'open_footer' ); // origin_open_footer ?>

			<div class="footer-content">
				
				<?php echo apply_atomic_shortcode( 'footer_content', hybrid_get_setting( 'footer_insert' ) ); ?>

				<?php do_atomic( 'footer' ); // origin_footer ?>

				<div id="language-buttons" style="float:right;">
					<button id="button-english" onclick="showEnglish()">English</button>
					<button id="button-spanish" onclick="showSpanish()">Spanish</button>
				</div>

			</div>

			<?php do_atomic( 'close_footer' ); // origin_close_footer ?>

		</div><!-- #footer -->

		<?php do_atomic( 'after_footer' ); // origin_after_footer ?>
		
		</div><!-- .wrap -->

	</div><!-- #container -->

	<?php do_atomic( 'close_body' ); // origin_close_body ?>

	<?php wp_footer(); // wp_footer ?>
	
</body>
</html>

header.php
<?php
/**
 * Header Template Without Breadcrumb
 *
 * The header template is generally used on every page of your site. Nearly all other templates call it 
 * somewhere near the top of the file. It is used mostly as an opening wrapper, which is closed with the 
 * footer.php file. It also executes key functions needed by the theme, child themes, and plugins. 
 *
 * This template does not show the "breadcrymb" div 
 *
 * @package Origin
 * @subpackage Template
 */
?>
<!doctype html>
<html <?php language_attributes(); ?>>
<head>
<meta http-equiv="Content-Type" content="<?php bloginfo( 'html_type' ); ?>; charset=<?php bloginfo( 'charset' ); ?>" />
<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    
<!-- Mobile viewport optimized -->
<meta name="viewport" content="width=device-width,initial-scale=1">

<?php if ( hybrid_get_setting( 'origin_favicon_url' ) ) { ?>
<!-- Favicon -->
<link rel="shortcut icon" href="<?php echo hybrid_get_setting( 'origin_favicon_url' ); ?>" />
<?php } ?>

<!-- Title -->
<title><?php hybrid_document_title(); ?></title>

<!-- Stylesheet -->	
<link rel="stylesheet" href="<?php echo get_stylesheet_uri(); ?>" type="text/css" />

<link rel="profile" href="http://gmpg.org/xfn/11" />
<link rel="pingback" href="<?php bloginfo( 'pingback_url' ); ?>" />

<!-- WP Head -->
<?php wp_head(); ?>

</head>

<!-- Javascript functions to choose the language of the page (English or Spanish) -->
<script type="text/javascript">

function showEnglish() {
	document.getElementById("spanish-text").style.display = "none";
	document.getElementById("english-text").style.display = "block";
}

function showSpanish() {
	document.getElementById("english-text").style.display = "none";
	document.getElementById("spanish-text").style.display = "block";
}
</script>

<body class="<?php hybrid_body_class(); ?>" onload="showSpanish()">

	<?php do_atomic( 'open_body' ); // origin_open_body ?>

	<div id="social-icons">
		<img src="wp-content/themes/origin/user-social-icons/twitter-icon.png" width="50" height="50" class="image" /> <br /><br />
		<img src="wp-content/themes/origin/user-social-icons/facebook-icon.png" width="50" height="50" class="image" /> <br /><br />
		<img src="wp-content/themes/origin/user-social-icons/google-icon.jpg" width="50" height="50" class="image" />
	</div>

	<div id="div-aux">
		<p><!-- div usado para la correcta posicion de los elementos de la pagina --></p>
	</div>

	<div id="container">

		<div class="wrap">

			<?php do_atomic( 'before_header' ); // origin_before_header ?>
	
			<div id="header">
	
				<?php do_atomic( 'open_header' ); // origin_open_header ?>
	
					<div id="branding">
						
						<?php origin_site_title(); ?>
						
					</div><!-- #branding -->
					
					<?php get_template_part( 'menu', 'primary' ); // Loads the menu-primary.php template. ?>

					<?php hybrid_site_description(); ?>


					<?php do_atomic( 'header' ); // origin_header ?>
	
				<?php do_atomic( 'close_header' ); // origin_close_header ?>
	
			</div><!-- #header -->
	
			<?php do_atomic( 'after_header' ); // origin_after_header ?>
	
			<?php do_atomic( 'before_main' ); // origin_before_main ?>
	
			<div id="main">
	
				<?php do_atomic( 'open_main' ); // origin_open_main ?>

admin/functions-admin.js
jQuery(document).ready(function($) {

    var formfield;
              
    // Upload image         
    $('#origin_favicon_upload_button, #origin_logo_upload_button').click(function() {
        formfield = $(this).prev().attr('id');
        tbframe_interval = setInterval(function() {
            $('#TB_iframeContent').contents().find('.savesend input[type="submit"]').val(js_text.insert_into_post);
        }, 500);        
        tb_show('', 'media-upload.php?post_id=0&type=image&TB_iframe=true');
        return false;
    });

    // Insert the image url into the input field
    window.original_send_to_editor = window.send_to_editor;
    window.send_to_editor = function(html) {  
        if (formfield) {      
            fileurl = $('img', html).attr('src');  
            $('#' + formfield).val(fileurl);     
            tb_remove();
            formfield = '';
        } else {
            window.original_send_to_editor(html);
        }
    } 

    // Colorpicker
    jQuery('#colorpicker_link_color').farbtastic('#origin_theme_settings-origin_link_color');
    
    jQuery('#origin_theme_settings-origin_link_color').blur( function() {
            jQuery('#colorpicker_link_color').hide();
    });
    
    jQuery('#origin_theme_settings-origin_link_color').focus( function() {
            jQuery('#colorpicker_link_color').show();
    });
             
});

admin/functions-admin.php
<?php
/*
 * Theme Settings
 *
 * @package Origin
 * @subpackage Template
 */
	
add_action( 'admin_menu', 'origin_theme_admin_setup' );

function origin_theme_admin_setup() {
    
	global $theme_settings_page;
	
	/* Get the theme settings page name */
	$theme_settings_page = 'appearance_page_theme-settings';

	/* Get the theme prefix. */
	$prefix = hybrid_get_prefix();

	/* Create a settings meta box only on the theme settings page. */
	add_action( 'load-appearance_page_theme-settings', 'origin_theme_settings_meta_boxes' );

	/* Add a filter to validate/sanitize your settings. */
	add_filter( "sanitize_option_{$prefix}_theme_settings", 'origin_theme_validate_settings' );
	
	/* Enqueue scripts */
	add_action( 'admin_enqueue_scripts', 'origin_admin_scripts' );
	
}

/* Adds custom meta boxes to the theme settings page. */
function origin_theme_settings_meta_boxes() {

	/* Add a custom meta box. */
	add_meta_box(
		'origin-theme-meta-box',			// Name/ID
		__( 'General settings', 'origin' ),	// Label
		'origin_theme_meta_box',			// Callback function
		'appearance_page_theme-settings',		// Page to load on, leave as is
		'normal',					// Which meta box holder?
		'high'					// High/low within the meta box holder
	);

	/* Add additional add_meta_box() calls here. */
}

/* Function for displaying the meta box. */
function origin_theme_meta_box() { ?>

	<table class="form-table">
	    
		<!-- Favicon upload -->
		<tr class="favicon_url">
			<th>
				<label for="<?php echo esc_attr( hybrid_settings_field_id( 'origin_favicon_url' ) ); ?>"><?php _e( 'Favicon:', 'origin' ); ?></label>
			</th>
			<td>
				<input type="text" id="<?php echo esc_attr( hybrid_settings_field_id( 'origin_favicon_url' ) ); ?>" name="<?php echo esc_attr( hybrid_settings_field_name( 'origin_favicon_url' ) ); ?>" value="<?php echo esc_url( hybrid_get_setting( 'origin_favicon_url' ) ); ?>" />
				<input id="origin_favicon_upload_button" class="button" type="button" value="Upload" />
				<br />
				<span class="description"><?php _e( 'Upload favicon image (recommended max size: 32x32).', 'origin' ); ?></span>
				
				<?php /* Display uploaded image */
				if ( hybrid_get_setting( 'origin_favicon_url' ) ) { ?>
                    <p><img src="<?php echo esc_url( hybrid_get_setting( 'origin_favicon_url' ) ); ?>" alt=""/></p>
				<?php } ?>
			</td>
		</tr>
		
		<!-- Logo upload -->
		<tr class="logo_url">
			<th>
				<label for="<?php echo esc_attr( hybrid_settings_field_id( 'origin_logo_url' ) ); ?>"><?php _e( 'Logo:', 'origin' ); ?></label>
			</th>
			<td>
				<input type="text" id="<?php echo esc_attr( hybrid_settings_field_id( 'origin_logo_url' ) ); ?>" name="<?php echo esc_attr( hybrid_settings_field_name( 'origin_logo_url' ) ); ?>" value="<?php echo esc_url( hybrid_get_setting( 'origin_logo_url' ) ); ?>" />
				<input id="origin_logo_upload_button" class="button" type="button" value="Upload" />
				<br />
				<span class="description"><?php _e( 'Upload logo image (recommended max width: 200px).', 'origin' ); ?></span>
				
				<?php /* Display uploaded image */
				if ( hybrid_get_setting( 'origin_logo_url' ) ) { ?>
                    <p><img src="<?php echo esc_url( hybrid_get_setting( 'origin_logo_url' ) ); ?>" alt=""/></p>
				<?php } ?>
			</td>
		</tr>		
		
		<!-- Font family -->
		<tr>
			<th>
				<label for="<?php echo esc_attr( hybrid_settings_field_id( 'origin_font_family' ) ); ?>"><?php _e( 'Font family:', 'origin' ); ?></label>
			</th>
			<td>
			    <select id="<?php echo esc_attr( hybrid_settings_field_id( 'origin_font_family' ) ); ?>" name="<?php echo esc_attr( hybrid_settings_field_name( 'origin_font_family' ) ); ?>">
				<option value="Bitter" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Bitter' ); ?>> <?php echo __( 'Bitter', 'origin' ) ?> </option>
				<option value="Georgia" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Georgia' ); ?>> <?php echo __( 'Georgia', 'origin' ) ?> </option>
				<option value="Droid Serif" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Droid Serif' ); ?>> <?php echo __( 'Droid Serif', 'origin' ) ?> </option>				
				<option value="Helvetica" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Helvetica' ); ?>> <?php echo __( 'Helvetica', 'origin' ) ?> </option>
				<option value="Istok Web" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Istok Web' ); ?>> <?php echo __( 'Istok Web', 'origin' ) ?> </option>
				<option value="Arial" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Arial' ); ?>> <?php echo __( 'Arial', 'origin' ) ?> </option>
				<option value="Verdana" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Verdana' ); ?>> <?php echo __( 'Verdana', 'origin' ) ?> </option>
				<option value="Lucida Sans Unicode" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Lucida Sans Unicode' ); ?>> <?php echo __( 'Lucida Sans Unicode', 'origin' ) ?> </option>
				<option value="Droid Sans" <?php selected( hybrid_get_setting( 'origin_font_family' ), 'Droid Sans' ); ?>> <?php echo __( 'Droid Sans', 'origin' ) ?> </option>
			    </select>
			</td>
		</tr>
		
		<!-- Font size -->
		<tr>
			<th>
			    <label for="<?php echo esc_attr( hybrid_settings_field_id( 'origin_font_size' ) ); ?>"><?php _e( 'Font size:', 'origin' ); ?></label>
			</th>
			<td>
			    <select id="<?php echo esc_attr( hybrid_settings_field_id( 'origin_font_size' ) ); ?>" name="<?php echo esc_attr( hybrid_settings_field_name( 'origin_font_size' ) ); ?>">
				<option value="16" <?php selected( hybrid_get_setting( 'origin_font_size' ), '16' ); ?>> <?php echo __( 'default', 'origin' ) ?> </option>
				<option value="17" <?php selected( hybrid_get_setting( 'origin_font_size' ), '17' ); ?>> <?php echo __( '17', 'origin' ) ?> </option>
				<option value="16" <?php selected( hybrid_get_setting( 'origin_font_size' ), '16' ); ?>> <?php echo __( '16', 'origin' ) ?> </option>
				<option value="15" <?php selected( hybrid_get_setting( 'origin_font_size' ), '15' ); ?>> <?php echo __( '15', 'origin' ) ?> </option>
				<option value="14" <?php selected( hybrid_get_setting( 'origin_font_size' ), '14' ); ?>> <?php echo __( '14', 'origin' ) ?> </option>				
				<option value="13" <?php selected( hybrid_get_setting( 'origin_font_size' ), '13' ); ?>> <?php echo __( '13', 'origin' ) ?> </option>
				<option value="12" <?php selected( hybrid_get_setting( 'origin_font_size' ), '12' ); ?>> <?php echo __( '12', 'origin' ) ?> </option>
			    </select>
			    <span class="description"><?php _e( 'The base font size in pixels.', 'origin' ); ?></span>
			</td>
		</tr>		
	    
		<!-- Link color -->
		<tr>
			<th>
				<label for="<?php echo esc_attr( hybrid_settings_field_id( 'origin_link_color' ) ); ?>"><?php _e( 'Link color:', 'origin' ); ?></label>
			</th>
			<td>
				<input type="text" id="<?php echo esc_attr( hybrid_settings_field_id( 'origin_link_color' ) ); ?>" name="<?php echo esc_attr( hybrid_settings_field_name( 'origin_link_color' ) ); ?>" size="8" value="<?php echo ( hybrid_get_setting( 'origin_link_color' ) ) ? esc_attr( hybrid_get_setting( 'origin_link_color' ) ) : '#dd5424'; ?>" data-hex="true" />
				<div id="colorpicker_link_color"></div>
				<span class="description"><?php _e( 'Set the theme link color.', 'origin' ); ?></span>
			</td>
		</tr>	    

		<!-- Custom CSS -->
		<tr>
			<th>
				<label for="<?php echo esc_attr( hybrid_settings_field_id( 'origin_custom_css' ) ); ?>"><?php _e( 'Custom CSS:', 'origin' ); ?></label>
			</th>
			<td>
				<textarea id="<?php echo esc_attr( hybrid_settings_field_id( 'origin_custom_css' ) ); ?>" name="<?php echo esc_attr( hybrid_settings_field_name( 'origin_custom_css' ) ); ?>" cols="60" rows="8"><?php echo wp_htmledit_pre( stripslashes( hybrid_get_setting( 'origin_custom_css' ) ) ); ?></textarea>
				<span class="description"><?php _e( 'Add your custom CSS here. It would overwrite any default or custom theme settings.', 'origin' ); ?></span>
			</td>
		</tr>

		<!-- End custom form elements. -->
	</table><!-- .form-table --><?php
	
}

/* Validate theme settings. */
function origin_theme_validate_settings( $input ) {
    
	$input['origin_favicon_url'] = esc_url_raw( $input['origin_favicon_url'] );
	$input['origin_logo_url'] = esc_url_raw( $input['origin_logo_url'] );
	$input['origin_font_family'] = wp_filter_nohtml_kses( $input['origin_font_family'] );
	$input['origin_font_size'] = wp_filter_nohtml_kses( $input['origin_font_size'] );
    $input['origin_link_color'] = wp_filter_nohtml_kses( $input['origin_link_color'] );      
    $input['origin_custom_css'] = wp_filter_nohtml_kses( $input['origin_custom_css'] );

    /* Return the array of theme settings. */
    return $input;
}

/* Enqueue scripts (and related stylesheets) */
function origin_admin_scripts( $hook_suffix ) {
    
    global $theme_settings_page;
	
    if ( $theme_settings_page == $hook_suffix ) {
	    
	    /* Enqueue Scripts */
	    wp_enqueue_script( 'origin_functions-admin', get_template_directory_uri() . '/admin/functions-admin.js', array( 'jquery', 'media-upload', 'thickbox', 'farbtastic' ), '1.0', false );

		/* Localize script strings */
		wp_localize_script( 'origin_functions-admin', 'js_text', array( 'insert_into_post' => __( 'Use this Image', 'origin' ) ) );	    
	    
	    /* Enqueue Styles */
	    wp_enqueue_style( 'origin_functions-admin', get_template_directory_uri() . '/admin/functions-admin.css', false, 1.0, 'screen' );
	    wp_enqueue_style( 'farbtastic' );
    }
}

?>

functions.php
<?php
/**
 * @package Origin
 * @subpackage Functions
 * @version 0.4
 * @author Galin Simeonov
 * @link http://alienwp.com
 * @license http://www.gnu.org/licenses/gpl-2.0.html
 */

/* Load the core theme framework. */
require_once( trailingslashit( TEMPLATEPATH ) . 'library/hybrid.php' );
$theme = new Hybrid();

/* Do theme setup on the 'after_setup_theme' hook. */
add_action( 'after_setup_theme', 'origin_theme_setup' );

/**
 * Theme setup function.  This function adds support for theme features and defines the default theme
 * actions and filters.
 *
 */
function origin_theme_setup() {

	/* Get action/filter hook prefix. */
	$prefix = hybrid_get_prefix();

	/* Add theme support for core framework features. */
	add_theme_support( 'hybrid-core-menus', array( 'primary' ) );
	add_theme_support( 'hybrid-core-sidebars', array( 'primary', 'subsidiary', 'after-singular' ) );
	add_theme_support( 'hybrid-core-widgets' );
	add_theme_support( 'hybrid-core-theme-settings', array( 'footer', 'about' ) );
	add_theme_support( 'hybrid-core-meta-box-footer' );
	add_theme_support( 'hybrid-core-shortcodes' );
	add_theme_support( 'hybrid-core-drop-downs' );
	add_theme_support( 'hybrid-core-template-hierarchy' );

	/* Add theme support for framework extensions. */
	add_theme_support( 'loop-pagination' );
	add_theme_support( 'get-the-image' );
	add_theme_support( 'cleaner-gallery' );
	add_theme_support( 'breadcrumb-trail' );

	/* Add theme support for WordPress features. */
	add_theme_support( 'automatic-feed-links' );

	/* Embed width/height defaults. */
	add_filter( 'embed_defaults', 'origin_embed_defaults' );

	/* Filter the sidebar widgets. */
	add_filter( 'sidebars_widgets', 'origin_disable_sidebars' );
        
	/* Image sizes */
	add_action( 'init', 'origin_image_sizes' );

	/* Excerpt ending */
	add_filter( 'excerpt_more', 'origin_excerpt_more' );
 
	/* Custom excerpt length */
	add_filter( 'excerpt_length', 'origin_excerpt_length' );    
        
	/* Filter the pagination trail arguments. */
	add_filter( 'loop_pagination_args', 'origin_pagination_args' );
	
	/* Filter the comments arguments */
	add_filter( "{$prefix}_list_comments_args", 'origin_comments_args' );	
	
	/* Filter the commentform arguments */
	add_filter( 'comment_form_defaults', 'origin_commentform_args', 11, 1 );
	
	/* Enqueue scripts (and related stylesheets) */
	add_action( 'wp_enqueue_scripts', 'origin_scripts' );
	
	/* Enqueue Google fonts */
	add_action( 'wp_enqueue_scripts', 'origin_google_fonts' );
	
	/* Style settings */
	add_action( 'wp_head', 'origin_style_settings' );
	
	/* Add the breadcrumb trail just after the container is open. */
	add_action( "{$prefix}_close_header", 'breadcrumb_trail' );
	
	/* Breadcrumb trail arguments. */
	add_filter( 'breadcrumb_trail_args', 'origin_breadcrumb_trail_args' );
	
	/* Add support for custom backgrounds */
	add_theme_support( 'custom-background' );
	
	/* Add theme settings */
	if ( is_admin() )
	    require_once( trailingslashit( TEMPLATEPATH ) . 'admin/functions-admin.php' );
	    
	/* Default footer settings */
	add_filter( "{$prefix}_default_theme_settings", 'origin_default_footer_settings' );
	
}

/**
 * Disables sidebars if viewing a one-column page.
 *
 */
function origin_disable_sidebars( $sidebars_widgets ) {
	
	global $wp_query;
	
	    if ( is_page_template( 'page-template-fullwidth.php' ) ) {
		    $sidebars_widgets['primary'] = false;
	    }

	return $sidebars_widgets;
}

/**
 * Overwrites the default widths for embeds.  This is especially useful for making sure videos properly
 * expand the full width on video pages.  This function overwrites what the $content_width variable handles
 * with context-based widths.
 *
 */
function origin_embed_defaults( $args ) {
	
	$args['width'] = 640;
		
	if ( is_page_template( 'page-template-fullwidth.php' ) )
		$args['width'] = 940;

	return $args;
}

/**
 * Excerpt ending 
 *
 */
function origin_excerpt_more( $more ) {	
	return '...';
}

/**
 *  Custom excerpt lengths 
 *
 */
function origin_excerpt_length( $length ) {
	return 40;
}

/**
 * Enqueue scripts (and related stylesheets)
 *
 */
function origin_scripts() {
	
	if ( !is_admin() ) {
		
		/* Enqueue Scripts */
		
		wp_register_script( 'origin_fancybox', get_template_directory_uri() . '/js/fancybox/jquery.fancybox-1.3.4.pack.js', array( 'jquery' ), '1.0', true );
		wp_enqueue_script( 'origin_fancybox' );
		
		wp_register_script( 'origin_fitvids', get_template_directory_uri() . '/js/fitvids/jquery.fitvids.js', array( 'jquery' ), '1.0', true );
		wp_enqueue_script( 'origin_fitvids' );
		
		wp_register_script( 'origin_footer-scripts', get_template_directory_uri() . '/js/footer-scripts.js', array( 'jquery', 'origin_fitvids', 'origin_fancybox' ), '1.0', true );
		wp_enqueue_script( 'origin_footer-scripts' );	

		/* Enqueue Styles */
		wp_enqueue_style( 'origin_fancybox-stylesheet', get_template_directory_uri() . '/js/fancybox/jquery.fancybox-1.3.4.css', false, 1.0, 'screen' );		
	}
}

/**
 * Pagination args 
 *
 */
function origin_pagination_args( $args ) {
	
	$args['prev_text'] = __( '&larr; Previous', 'origin' );
	$args['next_text'] = __( 'Next &rarr;', 'origin' );

	return $args;
}

/**
 *  Image sizes
 *
 */
function origin_image_sizes() {
	add_image_size( 'single-thumbnail', 636, 310, true );
}

/**
 *  Unregister Hybrid widgets
 *
 */
function origin_unregister_widgets() {
	
	unregister_widget( 'Hybrid_Widget_Search' );
	register_widget( 'WP_Widget_Search' );	
}

/**
 * Custom comments arguments
 * 
 */
function origin_comments_args( $args ) {
	
	$args['avatar_size'] = 50;
	return $args;
}

/**
 *  Custom comment form arguments
 * 
 */
function origin_commentform_args( $args ) {
	
	global $user_identity;

	/* Get the current commenter. */
	$commenter = wp_get_current_commenter();

	/* Create the required <span> and <input> element class. */
	$req = ( ( get_option( 'require_name_email' ) ) ? ' <span class="required">' . __( '*', 'origin' ) . '</span> ' : '' );
	$input_class = ( ( get_option( 'require_name_email' ) ) ? ' req' : '' );
	
	
	$fields = array(
		'author' => '<p class="form-author' . $input_class . '"><input type="text" class="text-input" name="author" id="author" value="' . esc_attr( $commenter['comment_author'] ) . '" size="40" /><label for="author">' . __( 'Name', 'origin' ) . $req . '</label></p>',
		'email' => '<p class="form-email' . $input_class . '"><input type="text" class="text-input" name="email" id="email" value="' . esc_attr( $commenter['comment_author_email'] ) . '" size="40" /><label for="email">' . __( 'Email', 'origin' ) . $req . '</label></p>',
		'url' => '<p class="form-url"><input type="text" class="text-input" name="url" id="url" value="' . esc_attr( $commenter['comment_author_url'] ) . '" size="40" /><label for="url">' . __( 'Website', 'origin' ) . '</label></p>'
	);
	
	$args = array(
		'fields' => apply_filters( 'comment_form_default_fields', $fields ),
		'comment_field' => '<p class="form-textarea req"><!--<label for="comment">' . __( 'Comment', 'origin' ) . '</label>--><textarea name="comment" id="comment" cols="60" rows="10"></textarea></p>',
		'must_log_in' => '<p class="alert">' . sprintf( __( 'You must be <a href="%1$s" title="Log in">logged in</a> to post a comment.', 'origin' ), wp_login_url( get_permalink() ) ) . '</p><!-- .alert -->',
		'logged_in_as' => '<p class="log-in-out">' . sprintf( __( 'Logged in as <a href="%1$s" title="%2$s">%2$s</a>.', 'origin' ), admin_url( 'profile.php' ), esc_attr( $user_identity ) ) . ' <a href="' . wp_logout_url( get_permalink() ) . '" title="' . esc_attr__( 'Log out of this account', 'origin' ) . '">' . __( 'Log out &rarr;', 'origin' ) . '</a></p><!-- .log-in-out -->',
		'comment_notes_before' => '',
		'comment_notes_after' => '',
		'id_form' => 'commentform',
		'id_submit' => 'submit',
		'title_reply' => __( 'Leave a Reply', 'origin' ),
		'title_reply_to' => __( 'Leave a Reply to %s', 'origin' ),
		'cancel_reply_link' => __( 'Click here to cancel reply.', 'origin' ),
		'label_submit' => __( 'Post Comment &rarr;', 'origin' ),
	);
	
	return $args;
}

/**
 * Breadcrumb trail arguments.
 *
 */
function origin_breadcrumb_trail_args( $args ) {

	$args['before'] = '';
	$args['separator'] = '&raquo;';
	$args['front_page'] = false;
	
	return $args;
}

/**
 * Default footer settings
 *
 */
function origin_default_footer_settings( $settings ) {
    
    $settings['footer_insert'] = '<p class="copyright">' . __( 'Copyright &#169; [the-year] [site-link]', 'origin' ) . '</p>' . "\n\n" . '<p class="credit">' . __( 'Powered by [wp-link] and [theme-link]', 'origin' ) . '</p>';
    
    return $settings;
}

/**
 * Google fonts
 *
 */
function origin_google_fonts() {
	
	if ( hybrid_get_setting( 'origin_font_family' ) ) {
		
		switch ( hybrid_get_setting( 'origin_font_family' ) ) {			
			case 'Bitter':
				wp_enqueue_style( 'font-bitter', 'http://fonts.googleapis.com/css?family=Bitter', false, 1.0, 'screen' );
				break;
			case 'Droid Serif':
				wp_enqueue_style( 'font-droid-serif', 'http://fonts.googleapis.com/css?family=Droid+Serif', false, 1.0, 'screen' );
				break;
			case 'Istok Web':
				wp_enqueue_style( 'font-istok-web', 'http://fonts.googleapis.com/css?family=Istok+Web', false, 1.0, 'screen' );
				break;
			case 'Droid Sans':
				wp_enqueue_style( 'font-droid-sans', 'http://fonts.googleapis.com/css?family=Droid+Sans', false, 1.0, 'screen' );
				break;
		}
	} else {
		wp_enqueue_style( 'font-bitter', 'http://fonts.googleapis.com/css?family=Bitter', false, 1.0, 'screen' );
	}
}

/**
 * Style settings
 *
 */
function origin_style_settings() {
	
	echo "\n<!-- Style settings -->\n";
	echo "<style type=\"text/css\" media=\"all\">\n";
	
	if ( hybrid_get_setting( 'origin_font_size' ) )
		echo 'html { font-size: ' . hybrid_get_setting( 'origin_font_size' ) . "px; }\n";
	
	if ( hybrid_get_setting( 'origin_font_family' ) )
		echo 'body { font-family: ' . hybrid_get_setting( 'origin_font_family' ) . ", serif; }\n";
	
	if ( hybrid_get_setting( 'origin_link_color' ) ) {
		echo 'a, a:visited, #footer a:hover, .entry-title a:hover { color: ' . hybrid_get_setting( 'origin_link_color' ) . "; }\n";
		echo '#respond #submit, .button, a.button, .wpcf7-submit, #loginform .button-primary { background-color: ' . hybrid_get_setting( 'origin_link_color' ) . "; }\n";
		echo "a:hover, a:focus { color: #000; }\n";
	}	
	if ( hybrid_get_setting( 'origin_custom_css' ) )
		echo hybrid_get_setting( 'origin_custom_css' ) . "\n";
	
	echo "</style>\n";

}

/**
 * Origin site title.
 * 
 */
function origin_site_title() {
	
	if ( hybrid_get_setting( 'origin_logo_url' ) ) {	
	
		$tag = ( is_front_page() ) ? 'h1' : 'div';	
			
		echo '<' . $tag . ' id="site-title">' . "\n";
			echo '<a href="' . get_home_url() . '" title="' . get_bloginfo( 'name' ) . '" rel="Home">' . "\n";
				echo '<img class="logo" src="' . esc_url( hybrid_get_setting( 'origin_logo_url' ) ) . '" alt="' . get_bloginfo( 'name' ) . '" />' . "\n";
			echo '</a>' . "\n";
		echo '</' . $tag . '>' . "\n";
	
	} else {
	
		hybrid_site_title();
	
	}
}

?>